envoy/grpc/async_client_manager.h

```diff
@@ -15,11 +17,21 @@ class AsyncClientFactory {
 
   /**
    * Create a gRPC::RawAsyncClient.
+   * Prefer AsyncClientManager::getOrCreateRawAsyncClient() to creating uncached raw async client
+   * from factory. Only call this method when the raw async client must be owned
+   * exclusively. For example, some filters pass *this reference to raw client. In this case, raw
```

Comment:
s/raw client/the client/

Can you give an example of this? How is the `*this` ref passed to the raw client?

Reply:
ExternalProcessorStreamImpl::ExternalProcessorStreamImpl(
    Grpc::AsyncClient<ProcessingRequest, ProcessingResponse>&& client,
    ExternalProcessorCallbacks& callbacks)
    : callbacks_(callbacks) {
  client_ = std::move(client);
  auto descriptor = Protobuf::DescriptorPool::generated_pool()->FindMethodByName(kExternalMethod);
  Http::AsyncClient::StreamOptions options;
  stream_ = client_.start(*descriptor, *this, options);
}

Reply:
Cause memory access fault in some test.

Reply:
I may not be understanding...do you mean the async client is moved into the filter object? Yeah that won't work. We'd have to change the use-case to store it in the filter object via the shared ptr.

Reply:
No, reference to the filter object is stored in async stream, when the async stream outlives the filter, it will have dangling reference.